### ISSUE TITLE
Refactor GetNext to use local state

### ISSUE THREAD

**Issue by timpeat** [ministryofjustice/laa-review-criminal-legal-aid#161]
https://github.com/ministryofjustice/laa-review-criminal-legal-aid/blob/main/app/models/get_next.rb

**Comment by timpeat** [comment]
TLDR: This issue was closed because a different approach to local state was chosen, which would not bring an advantage to the "get next" algorithm.

Initially, a different approach to maintaining local state was considered for the following reasons:

1. To reduce API requests
2. To reduce the likelihood of the same application being proposed as the next application for multiple caseworkers
3. To simplify the code

The "get next" algorithm currently searches the datastore API for the oldest unassigned applications. When this issue was raised, it was thought that creating a crime application read model in the review database would allow current applications to be queried locally. The idea was that this read model would be updated by a handler listening to the ApplicationReceived events.

However, after it was decided that the Maat integration would not use SNS, it was determined that making SNS/SQS a critical architectural dependency was not justifiable for the review state issue alone. As a result, a more relaxed approach to maintaining local state was chosen. If an SNS queue is set up, the review will actively maintain a local understanding of new applications. If an SNS queue is not set up, then the review local state is only aware of new applications when a user accesses the service.

This has several advantages, including:

1. Reduced developer setup, as the SNS is not required in development
2. Handling of any failed message delivery, because the system is designed to work without SNS
3. SNS is not a critical architectural dependency

However, this more relaxed approach would not be suitable for the "get next" algorithm, which needs to know about new applications in real-time. The reasons for considering using local state for "get next" do not justify changing the relaxed approach to maintaining local state.

Transactional consistency, preventing an application from being assigned to two caseworkers, is already enforced by the assigning aggregate. Although there is a risk of duplicate applications being offered to multiple caseworkers, the assigning command will not succeed. The UI should handle this case, and we can monitor to see if this is a significant problem.
